GeckoView Nightly manual update required: onFilePrompt

> > Task :browser-engine-gecko-nightly:compileDebugKotlin FAILED
> e: /build/android-components/components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegate.kt: (117, 5): 'onFilePrompt' overrides nothing
> FAILURE: Build failed with an exception.

https://taskcluster-ui.herokuapp.com/tasks/QQpsPCV3R_2zrLd3KmJFog/runs/0/logs/https%3A%2F%2Fqueue.taskcluster.net%2Fv1%2Ftask%2FQQpsPCV3R_2zrLd3KmJFog%2Fruns%2F0%2Fartifacts%2Fpublic%2Flogs%2Flive.log

--- a/buildSrc/src/main/java/Gecko.kt
+++ b/buildSrc/src/main/java/Gecko.kt
@@ -6,7 +6,7 @@ internal object GeckoVersions {
     /**
      * GeckoView Nightly Version.
      */
-    const val nightly_version = "70.0.20190730095207"
+    const val nightly_version = "70.0.20190731215544"
 
     /**
      * GeckoView Beta Version.

--- a/components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegate.kt
+++ b/components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegate.kt
@@ -38,6 +38,9 @@ import org.mozilla.geckoview.GeckoSession.PromptDelegate.BUTTON_TYPE_NEGATIVE
 import org.mozilla.geckoview.GeckoSession.PromptDelegate.BUTTON_TYPE_NEUTRAL
 import org.mozilla.geckoview.GeckoSession.PromptDelegate.BUTTON_TYPE_POSITIVE
 import org.mozilla.geckoview.GeckoSession.PromptDelegate.ButtonCallback
+import org.mozilla.geckoview.GeckoSession.PromptDelegate.CAPTURE_TYPE_ANY
+import org.mozilla.geckoview.GeckoSession.PromptDelegate.CAPTURE_TYPE_ENVIRONMENT
+import org.mozilla.geckoview.GeckoSession.PromptDelegate.CAPTURE_TYPE_USER
 import org.mozilla.geckoview.GeckoSession.PromptDelegate.Choice.CHOICE_TYPE_MENU
 import org.mozilla.geckoview.GeckoSession.PromptDelegate.Choice.CHOICE_TYPE_MULTIPLE
 import org.mozilla.geckoview.GeckoSession.PromptDelegate.Choice.CHOICE_TYPE_SINGLE
@@ -114,11 +117,13 @@ internal class GeckoPromptDelegate(private val geckoEngineSession: GeckoEngineSe
         }
     }
 
+    @Suppress("LongMethod")
     override fun onFilePrompt(
         session: GeckoSession,
         title: String?,
         selectionType: Int,
         mimeTypes: Array<out String>?,
+        capture: Int,
         callback: FileCallback
     ) {
 
@@ -132,7 +137,12 @@ internal class GeckoPromptDelegate(private val geckoEngineSession: GeckoEngineSe
 
         val isMultipleFilesSelection = selectionType == GeckoSession.PromptDelegate.FILE_TYPE_MULTIPLE
 
-        val captureMode = PromptRequest.File.FacingMode.NONE
+        val captureMode = when (capture) {
+            CAPTURE_TYPE_ANY -> PromptRequest.File.FacingMode.ANY
+            CAPTURE_TYPE_USER -> PromptRequest.File.FacingMode.FRONT_CAMERA
+            CAPTURE_TYPE_ENVIRONMENT -> PromptRequest.File.FacingMode.BACK_CAMERA
+            else -> PromptRequest.File.FacingMode.NONE
+        }
 
         val onSelectSingle: (Context, Uri) -> Unit = { context, uri ->
             callback.confirm(context, uri.toFileUri(context))

--- a/components/browser/engine-gecko-nightly/src/test/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegateTest.kt
+++ b/components/browser/engine-gecko-nightly/src/test/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegateTest.kt
@@ -20,8 +20,8 @@ import mozilla.components.concept.engine.prompt.PromptRequest.MultipleChoice
 import mozilla.components.concept.engine.prompt.PromptRequest.SingleChoice
 import mozilla.components.support.ktx.kotlin.toDate
 import mozilla.components.support.test.mock
-import org.junit.Assert
 import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
 import org.junit.Assert.assertNotNull
 import org.junit.Assert.assertTrue
 import org.junit.Test
@@ -662,7 +662,14 @@ class GeckoPromptDelegateTest {
             }
         })
 
-        promptDelegate.onFilePrompt(mock(), "title", GeckoSession.PromptDelegate.FILE_TYPE_SINGLE, emptyArray(), callback)
+        promptDelegate.onFilePrompt(
+            mock(),
+            "title",
+            GeckoSession.PromptDelegate.FILE_TYPE_SINGLE,
+            emptyArray(),
+            GeckoSession.PromptDelegate.CAPTURE_TYPE_NONE,
+            callback
+        )
         assertTrue(request is PromptRequest.File)
 
         val filePickerRequest = request as PromptRequest.File
@@ -677,14 +684,20 @@ class GeckoPromptDelegateTest {
         assertTrue(onDismissWasCalled)
 
         assertTrue(filePickerRequest.mimeTypes.isEmpty())
-        Assert.assertFalse(filePickerRequest.isMultipleFilesSelection)
+        assertFalse(filePickerRequest.isMultipleFilesSelection)
+        assertEquals(PromptRequest.File.FacingMode.NONE, filePickerRequest.captureMode)
 
         promptDelegate.onFilePrompt(
-            mock(), "title",
-            GeckoSession.PromptDelegate.FILE_TYPE_MULTIPLE, emptyArray(), callback
+            mock(),
+            "title",
+            GeckoSession.PromptDelegate.FILE_TYPE_MULTIPLE,
+            emptyArray(),
+            GeckoSession.PromptDelegate.CAPTURE_TYPE_USER,
+            callback
         )
 
         assertTrue((request as PromptRequest.File).isMultipleFilesSelection)
+        assertEquals(PromptRequest.File.FacingMode.FRONT_CAMERA, (request as PromptRequest.File).captureMode)
     }
 
     @Test

